middleware/mixins/dereference.js

```diff
@@ -1,45 +1,125 @@
-const jsonld = require('jsonld');
-const { defaultContext } = require('@semapps/core');
+const isObject = require('isobject');
 
+/**
+ * @description MoleculerJS mixin to be applied on the ControlledContainerMixin within a Semapps project.
+ * Provides resource dereferencing after ldp.resource.get execution
+ * 
+ * @module MoleculerJS Mixin - Dereference
+ * 
+ * Features include:
+ * - `get`: Retrieve a specific property from the main data.
+ * - `dereference`: Recursively dereference properties from the main data.
+ * - `handleAfterGet`: Post-retrieval result processing applying a dereferencing plan.
+ * 
+ * The mixin also introduces an `after.get` hook for automated processing following each retrieval action.
+ * 
+ * Usage:
+ * This mixin requires the presence of an `ldp.resource.get` service within the MoleculerJS environment for 
+ * effective LDP data interaction, emphasizing the need for a supportive ecosystem that includes necessary 
+ * LDP resource management capabilities.
+ * 
+ * Example moleculer service:
+ * ```javascript
+ * module.exports = {
+ * name: 'resources',
+ * mixins: [ControlledContainerMixin, DereferenceMixin],
+ * settings: {
+ *   path: '/resources',
+ *   acceptedTypes: ['prefix:Classe'],
+ *   preferredView: '/Resource',
+ *   dereferencePlan: [
+ *     {
+ *       "p": "schema:member",
+ *       "n": [
+ *         { "p": "schema:affiliation" }
+ *       ]
+ *     }
+ *   ]
+ *  }
+ * }
+ * ```
+ */
 
 module.exports = {
     methods: {
-        async handleAfterGet(ctx, res) {  
-            const ldpNavigator = new LDPNavigator(); // To resolve predicates asked in dereferencePlan without using data in memory, it's necessary to create a new instance of LDPNavigator. Dereferencing the relation won't fetch the URL unless LDPNavigator is a new instance.
 
-            const dereferencePlan = this.settings.dereferencePlan || [];
-            const headers = {
-                'accept': 'application/ld+json',
-            };
-          
-            // Check authorization for get request
-            if (ctx.meta.headers && ctx.meta.headers.authorization) {
-            headers.authorization = ctx.meta.headers.authorization;
-            }
-          
-            ldpNavigator.setAdapters([
-                new FetchAdapter({
-                    headers: headers
-                })
-            ]);
+        /**
+        * Get a property from main data. This is a wrapper around ldp.resource.get to allow us to pass in the property to the API
+        * 
+        * @param ctx - moleculer context
+        * @param mainData - The main data to get the property from
+        * @param property - The property to get from mainData. It must be a property of mainData
+        * 
+        * @return { Promise } The result of the get operation as a JSON object with @context removed from the result
+        */
+        async get(ctx,mainData, property) {
```

Comment:
If all you do with these 2 params is putting them together (`mainData[property]`), why not pass a single `resourceUri` param ? It will be more clear.
I would also rename the method to `getWithoutContext` so it is clear what it is doing.